designs/helper_to_upgrade_projects_by_rescaffolding.md

```diff
@@ -0,0 +1,155 @@
+version/plugin and then, re-add all code source on top again. Therefore, this command would help users in
+order to minimize the required effort by automating which shows technically possible of this process.
+
+The main motivation of this proposal is to provide a helper for upgrades and
+make less painful this process. Examples:
+
+- See the discussion [How to regenerate scaffolding?](https://github.com/kubernetes-sigs/kubebuilder/discussions/2864)
+- From [slack channel By Paul Laffitte](https://kubernetes.slack.com/archives/CAR30FCJZ/p1675166014762669)
+
+### Goals
+
+- Help users upgrade their project with the latest changes
+- Help users to re-scaffold the projects from the scratch based on what was done previously with the tool
+- Make less painful the process to upgrade
+
+### Non-Goals
+
+- Change the default layout or how the KubeBuilder CLI works
+- Deal with customizations or deviations from the proposed layout
+- Be able to perform the project upgrade to the latest changes without human bean interactions
+- Deal and support external plugins
+- Provides support to [declarative](https://book.kubebuilder.io/plugins/declarative-v1.html) plugin
+  since it is desired and planned to decouple this solution and donate this plugin to its own authors [More info](https://github.com/kubernetes-sigs/kubebuilder/issues/3186)
+- Provide support old version prior we have the Project config (Kubebuilder < 3x) and the go/v2 layout which exist to ensure  a backwards compatibility with legacy layout provide by Kubebuilder 2x
+
+## Proposal
+
+The proposed solution to achieve this goal is to create an alpha command as described
+in the example section above, see:
+
+```shell
+kubebuilder alpha generate \
+    --from-project-file=<path> 
+    --to=<path where the project should be re-scaffold> 
+    --backup=<path-where the current version of the project should be copied as backup>
+    --layout-plugins=<chain of plugins key that can be used to create the layout with init sub-command>
+```
+
+Where:
+
+- project-file: If not informed then, the command would check it in the current directory
+- output: If not informed then, it should be the current repository
+- backup: if not informed then, it would be the current path -1 level with the name backup
+- layout-plugins: if not informed then, it is the same plugin chain available in the layout field
+
+This command would mainly perform the following operations:
+
+- 1. Make a backup of the current project
+- 2. Ensure that the output path is clean
+- 3. Read the [PROJECT config][project-config]
+- 4. Re-run all commands using the KubeBuilder binary to recreate the project in the output directory
+
+### User Stories
+
+- As a developer I can regenerate my project from the scratch based on all commands that I used the tool to build
+  my project previously, so that I can easily upgrade my current project to new CLI/plugin versions and get the
+  latest changes, bug fixes and features
```

Comment:
Suggestion:
> As an operator author, I can re-generate my project from the scratch by automating the commands based on my previous input to the project, so that I can easily migrate my project to the new layout using the newer CLI/plugin versions, which supports the latest changes, bug fixes and features.